src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs

```diff
@@ -76,9 +82,20 @@ internal class IssueHandler
                 // The text buffer might have changed since the analysis was triggered, so translate
                 // all issues to the current snapshot.
                 // See bug #1487: https://github.com/SonarSource/sonarlint-visualstudio/issues/1487
-                var translatedIssues = translateSpans(issues, textDocument.TextBuffer.CurrentSnapshot);
-
-                var newSnapshot = new IssuesSnapshot(projectName, projectGuid, textDocument.FilePath, translatedIssues);
+                var translatedIssues = translateSpans(issues, textDocument.TextBuffer.CurrentSnapshot).ToList();
```

Comment:
Please add review comments to help reviewers understand why changes have been made - what seems obvious to you as the person who made the change might not be obvious to anyone else reading it.

Here, I'm guessing you added the `ToList()` because you are filtering the collection twice before. Using `ToArray()` would be cheaper as it's a lighter-weight data structure.

However, is it required? Doesn't `DoTranslateSpans` return an array already?

Reply:
Here ToList is because we enumerate twitce afterwards: one for hotspots, one for issues

Reply:
But yes, I missed that translatespans already returns an array. Will fix the method